fedimint-core/src/module/mod.rs

```diff
@@ -155,7 +155,7 @@ impl ApiError {
 /// State made available to all API endpoints for handling a request
 pub struct ApiEndpointContext<'dbtx> {
     db: Database,
-    dbtx: DatabaseTransaction<'dbtx>,
+    dbtx: DatabaseTransaction<'dbtx, Committable>,
     has_auth: bool,
     request_auth: Option<ApiAuth>,
 }
```

Comment:
A longer term goal is getting rid of writable DBTXns in API endpoints to avoid writing non-consensus data to consensus-relevant DB columns. One way of implementing this would be supplying a non-committable consensus DBTX and a committable non-consensus DBTX that uses another prefix.